### PR TITLE
feat(federations): provide admin setting to enable Federation in Talk

### LINKED ISSUE #11279: 🌐 Federated invitation handling 💌 - Frontend
- Overview at https://github.com/nextcloud/spreed/issues/11229

---

- [x] Administration settings / Talk / Federations:
  - [x] Enable federations in Talk for this instance
  - [x] Allow users on this server to invite users to Talk conversations from all other servers / only trusted servers / only specific groups
- [x] Participants Tab:
  - [x] List remote users provided by `'core/autocomplete/get'` endpoint (should be there, if `settings/admin/sharing` is [configured](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/federated_cloud_sharing_configuration.html) and other servers provide their users
   - [x] Allow users on this server to invite users to Talk conversations from other servers (works if regex is matching `userid@domain`)
- [x] App level:
  - [x] invite - join conversation when invitation is accepted
    - [x] update participants list on moderator side from system message
  - [x] dismiss -verify that works as expected
  - [x] disinvite - verify that conversation is removed, participant is left and store is purged as for normal disinvite
  - [x] list - get all pending invitations from remore servers, and list them

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Part of #11279
  * Add Admin setting section for Federation (in debug mode only)

### 🖼️ Screenshots / Screencasts

image

### 🚧 Tasks

- [x] Check wording

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

---

### 🚧 Tasks

- [ ] Check code changes

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not possible
- [ ] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/AdminSettings/Federations.vue
Comment (nickvergessen) [inline src/components/AdminSettings/Federations.vue:26]: The thing is called "Federation" (singular there  is no plural of it in English)

**Review by nickvergessen — COMMENTED** [review]
Should at least hide it when not on debug mode, so we don't forget about it.

Or we default federation to enabled when debug is on, then you also don't need the setting for now, right?

**Comment by Antreesy** [comment]
> Or we default federation to enabled when debug is on, then you also don't need the setting for now, right?

Made changes, now it's visible in debug mode only
I'd keep in configurable anyway. For now it's a good base for putting here new settings, if they'll arise one day.

File: src/App.vue
Comment (nickvergessen) [inline src/App.vue:545]: unrelated?

Comment (Antreesy) [reply src/App.vue:545]: Supposed to be there for handling notification actions like "Accept invitation", to redirect a user to federated conversation, but API doesn't return a token for that (as we discussed it earlier)

Comment (nickvergessen) [reply src/App.vue:545]: but this PR is about the admin setting, not about notifications? :D

Comment (DorraJaouad) [reply src/App.vue:545]: So are you just preparing the handling before the API is ready? you can just stash it or will you wait until API PR is created and merged?

Comment (nickvergessen) [reply src/App.vue:545]: exactly what Dorra said. The second commit should be a separate PR :)

Comment (Antreesy) [reply src/App.vue]: Fiine

**Review by DorraJaouad — APPROVED** [review]
Tested and wording is fine.

**Pull request merged**
